platform/engineering/design-docs/2022-03-11-schedule-pending-appt.md

```diff
@@ -0,0 +1,156 @@
+
+The intended audience is for the engineering team.
+
+### Background
+Currently, the `vets-website` scheduling feature is as follows:  
+
+- Verify user is registered at a facility
+   - `/v0/user`
+- Pick type of care to schedule an appointment for
+   - Check community care eligibility  
+   - `/vaos/v0/community_care/eligibility/{type of care}`
+   - Receive true or false for type of care
+- If type of care is eligible, user can select to make appointment request at VA facility or community cares facility
+- User is given list of facilities to choose from
+  - Get facility data 
+  - `/vaos/v2/facilities?children=true&ids[]={facility id}&ids[]={facility id}`
+  - Note: facility id's come from registered facilities in `/v0/user`
+  - Get additional facility scheduling information such as services provided and if available to schedule request (including child facilities)
+  - `/vaos/v2/scheduling/configurations?facility_ids[]={facility_id}&facility_ids[]={facility_id}`
+- User chooses facility
+  - Check if able to schedule request at given facility and service 
+  - `/vaos/v2/eligibility?facility_id={id}&clinical_service_id={service_type}&type=request`
+  - Multiple checks and responses are done here:
+      - Is this a Cerner facility? if so, return cerner portal link
+      - Is primary facility? if not, ensure veteran has had visit within 12-24 months to proceed
+      - Is facility allow scheduled requests?, 
+      - Is request limit reached?
+- If Community Cares facility, choose provider
+  - `/facilities_api/v1/ccp/provider?{coordinate information}`
+- Choose desired date and times then schedule appointment request and post appointment request
+  - `/vaos/v2/appointments` 
+
+### High Level Design
+Implementation of this feature will largely follow the logic listed in the background with a key difference to improve user experience. all consumed APIS will be aggregated at the beginning of the process in order to create a single new endpoint to provide all information needed to schedule a request. It will only list types of care that are supported, along with the facilities that support that type of service. This will benefit the process in multiple ways:
+   - Move any business logic from the front end to the back end 
+   - Simplify overall backend and front complexity. 
+   - Create a better UX experience. Users will know of a service and facility eligibility before selecting it and waiting for a response.  
+
+## Specifics
+
+### Detailed Design
+The backend business logic will mirror the process explained in the background above except rather then only querying the selected types of service and facilities, all registered facilities will be queried to create a comprehensive list. If load times get unacceptably long due to front loading all queries, we will explore pre-caching results. 
```

Comment:
```suggestion
The backend business logic will mirror the process explained in the background above except rather than only querying the selected service and facilities, we will query all registered facilities to create a comprehensive list. If load times get unacceptably long due to front loading all queries, we will explore pre-caching results.
```